src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -126,6 +129,81 @@ public GuildImpl(JDAImpl api, long id)
             memberPresences = null;
     }
 
+    public void invalidate()
+    {
+        //Remove everything from global cache
+        // this prevents some race-conditions for getting audio managers from guilds
+        SnowflakeCacheViewImpl<Guild> guildView = getJDA().getGuildsView();
+        SnowflakeCacheViewImpl<StageChannel> stageView = getJDA().getStageChannelView();
+        SnowflakeCacheViewImpl<TextChannel> textView = getJDA().getTextChannelsView();
+        SnowflakeCacheViewImpl<ThreadChannel> threadView = getJDA().getThreadChannelsView();
+        SnowflakeCacheViewImpl<NewsChannel> newsView = getJDA().getNewsChannelView();
+        SnowflakeCacheViewImpl<VoiceChannel> voiceView = getJDA().getVoiceChannelsView();
+        SnowflakeCacheViewImpl<Category> categoryView = getJDA().getCategoriesView();
+
+        guildView.remove(id);
+
+        try (UnlockHook hook = stageView.writeLock())
+        {
+            getStageChannelCache()
+            .forEachUnordered(chan -> stageView.getMap().remove(chan.getIdLong()));
```

Comment:
looks like this is missing a bit of indentation
```suggestion
            getStageChannelCache()
                .forEachUnordered(chan -> stageView.getMap().remove(chan.getIdLong()));
```